lib/time.js

```diff
@@ -349,6 +415,219 @@ function CronTime(luxon) {
+			const startingMinute = (beforeJumpingPoint.minute + 1) % 60;
+			const startingHour =
+				(beforeJumpingPoint.hour + (startingMinute === 0)) % 24;
+
+			const hourRangeSize = afterJumpingPoint.hour - startingHour + 1;
+			const isHourJump = startingMinute === 0 && afterJumpingPoint.minute === 0;
+
+			// There exist DST jumps other than 1 hour long, and the function is built to deal with it.
+			// It may be overkill to assume some cases, but it shouldn't cost much at runtime.
+			// https://en.wikipedia.org/wiki/Daylight_saving_time_by_country
+			if (hourRangeSize === 2 && isHourJump) {
+				// Exact 1 hour jump, most common real-world case.
+				// There is no need to check minutes and seconds, as any value would suffice.
+				return startingHour in this.hour;
+			} else if (hourRangeSize === 1) {
+				// less than 1 hour jump, rare but does exist.
+				return (
+					startingHour in this.hour &&
+					this._checkTimeInSkippedRangeSingleHour(
+						startingMinute,
+						afterJumpingPoint.minute
+					)
+				);
+			} else {
+				// non-round or multi-hour jump. (does not exist in the real world at the time of writing)
+				return this._checkTimeInSkippedRangeMultiHour(
+					startingMinute,
+					afterJumpingPoint.minute,
+					startingHour,
+					afterJumpingPoint.hour
+				);
+			}
+		},
+
+		/**
+		 * Component of checking if a CronJob time existed in a DateTime range skipped by DST.
+		 * This subroutine makes a further assumption that the skipped range is fully contained in one hour,
+		 * and that all other larger units are valid for the job.
+		 *
+		 * for example a jump from 02:00:00 to 02:30:00, but not from 02:00:00 to 03:00:00.
+		 * @see _checkTimeInSkippedRange
+		 *
+		 * This is done by checking if any minute in startMinute - endMinute is valid, excluding endMinute.
+		 * For endMinute, there is only a match if the 0th second is a valid time.
+		 */
+		_checkTimeInSkippedRangeSingleHour: function (startMinute, endMinute) {
+			for (let minute = startMinute; minute < endMinute; ++minute) {
+				if (minute in this.minute) return true;
+			}
+
+			// Unless the very last second of the jump matched, there is no match.
+			return endMinute in this.minute && 0 in this.second;
+		},
+
+		/**
+		 * Component of checking if a CronJob time existed in a DateTime range skipped by DST.
+		 * This subroutine assumes the jump touches at least 2 hours, but the jump does not necessarily fully contain these hours.
+		 *
+		 * @see _checkTimeInSkippedRange
+		 *
+		 * This is done by defining the minutes to check for the first and last hour,
+		 * and checking all 60 minutes for any hours in between them.
+		 *
+		 * If any hour x minute combination is a valid time, true is returned.
+		 * The endMinute x endHour combination is only checked with the 0th second, since the rest would be out of the range.
+		 */
+		_checkTimeInSkippedRangeMultiHour: function (
+			startMinute,
+			endMinute,
+			startHour,
+			endHour
```

Comment:
personally the order `startHour`, `startMinute`, `endHour`, `endMinute` makes more sense in my brain

Reply:
Great suggestion, Hours last is pretty strange to begin with as well. Changed this to be `startHour, startMinute, endHour, endMinute` 👍

Reply:
so much easier to read in the test! thanks